src/sql/workbench/parts/notebook/outputs/notebookMarkdown.ts

```diff
@@ -0,0 +1,240 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+
+import { URI } from 'vs/base/common/uri';
+
+import { IDisposable, dispose } from 'vs/base/common/lifecycle';
+import { RenderOptions } from 'vs/base/browser/htmlContentRenderer';
+import { IMarkdownString, removeMarkdownEscapes } from 'vs/base/common/htmlContent';
+import { IMarkdownRenderResult } from 'vs/editor/contrib/markdown/markdownRenderer';
+import marked = require('vs/base/common/marked/marked');
+import { defaultGenerator } from 'vs/base/common/idGenerator';
+import { revive } from 'vs/base/common/marshalling';
+import * as fs from 'fs';
+
+export class NotebookMarkdownRenderer {
+	private _notebookURI: URI;
+	private _baseUrls: string[] = [];
+
+	constructor() {
+	}
+
+	render(markdown: IMarkdownString): IMarkdownRenderResult {
+		let disposables: IDisposable[] = [];
+		const element: HTMLElement = markdown ? this.renderMarkdown(markdown, undefined) : document.createElement('span');
+		return {
+			element,
+			dispose: () => dispose(disposables)
+		};
+	}
+
+	createElement(options: RenderOptions): HTMLElement {
+		const tagName = options.inline ? 'span' : 'div';
+		const element = document.createElement(tagName);
+		if (options.className) {
+			element.className = options.className;
+		}
+		return element;
+	}
+
+	parse(text: string): any {
+		let data = JSON.parse(text);
+		data = revive(data, 0);
+		return data;
+	}
+
+	/**
+	 * Create html nodes for the given content element.
+	 * Adapted from htmlContentRenderer. Ensures that the markdown renderer
+	 * gets passed in the correct baseUrl for the notebook's saved location,
+	 * respects the trusted state of a notebook, and allows command links to
+	 * be clickable.
+	 */
+	renderMarkdown(markdown: IMarkdownString, options: RenderOptions = {}): HTMLElement {
+		const element = this.createElement(options);
+
+		// signal to code-block render that the element has been created
+		let signalInnerHTML: () => void;
+		const withInnerHTML = new Promise(c => signalInnerHTML = c);
+
+		let notebookFolder = path.join(path.dirname(this._notebookURI.fsPath), path.sep);
```

Comment:
On Mac the separator is `/` but on Windows `\`. My understanding is that in Markdown all links should use URI format (e.g. `./my/relative/file.txt`). I can see supporting `\` on Windows, but we at least need to support `/`. I'd suggest we do that by re-converting the directory to a URI and getting the path from that?

Reply:
This isn't in markdown though. This is the root directory that we're passing into the markdown renderer, which will spit out links in the correct format. If I'm understanding you correctly, I think this is ok?

Reply:
As long as this works on Windows and mac, it's OK. I'm just trying to make sure that we don't get weirdly formed links like `C:\my\folder/relative/link/forward.txt`

Reply:
I won't block on this given your explanation. Let's just test / verify this.

Reply:
Ok I reworked this a tiny bit based on your feedback and from testing.

Here's what now works on Windows:
![tux1.0](tux.png)
![tux2.0](in/tuxinside.png) 
![tux3.0](/Users/chlafren/Documents/tux.png)
![tux4.0](C:\Users\chlafren\Documents\tux.png)

And on Mac
![tux](./tux.png)
![tux2.0](tux.png)
![tux3.0](/tux.png)
![tux4.0](/Users/chlafren/Documents/tux.png)